### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by vivek-gofynd**

### CODE REVIEW

File: extension.context.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (vivek-gofynd) [inline extension.context.json]: @jinalviranii Please remove this file

File: index.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
Comment (vivek-gofynd) [inline index.js]: Let's have one common server.js file only.

File: jsconfig.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (vivek-gofynd) [inline jsconfig.json]: Are we utilizing this in boilerplate ?
If not please remove it. @jinalviranii

File: next.config.mjs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  images: {
+    domains: ['cdn.fynd.com'], // Add your external domain here
```
Comment (vivek-gofynd) [inline next.config.mjs]: Please add proper comment here

File: dockerfile [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Use the official Node.js image.
+FROM node:18-alpine
```
Comment (vivek-gofynd) [inline dockerfile]: Please use latest LTS version

File: server.js [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+        event_type: 'delete',
+        version: '1',
+    },
+]
+
+// Initialize Express app
+const app = express();
+
+// In-memory storage for the access token
+let accessToken = null;
```
Comment (vivek-gofynd) [inline server.js]: Please use SQLlight to keep this consistent across all the boilerplates

File: src/app/globals.css [hunk 7]
```diff
@@ -0,0 +1,42 @@
+:root {
```
Comment (vivek-gofynd) [inline src/app/globals.css]: Let's keep only single CSS file across the project

File: src/app/layout.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import localFont from "next/font/local";
```
Comment (vivek-gofynd) [inline src/app/layout.js]: Don't need custom fonts support here

File: src/app/layout.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import "./globals.css";
+
+const geistSans = localFont({
```
Comment (vivek-gofynd) [inline src/app/layout.js]: Check if we can remove this file and move this to common page level

File: src/app/page.module.css [hunk 10]
```diff
@@ -0,0 +1,165 @@
+.page {
```
Comment (vivek-gofynd) [inline src/app/page.module.css]: Single CSS file

File: src/pages/style/index.css [hunk 11]
```diff
@@ -0,0 +1,151 @@
+html {
```
Comment (vivek-gofynd) [inline src/pages/style/index.css]: single css file

File: src/tests/__tests__/[...params].test.js [hunk 12]
```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import { render, waitFor } from '@testing-library/react';
+import HomePage, { getServerSideProps } from '../../pages/company/[...params]';
```
Comment (vivek-gofynd) [inline src/tests/__tests__/[...params].test.js]: simply add in tests folder remove__tests__

File: extension.context.json (hunk 1 above)
Comment (jinalviranii) [reply extension.context.json]: Done

File: next.config.mjs (hunk 4 above)
Comment (jinalviranii) [reply next.config.mjs]: done

File: jsconfig.json (hunk 3 above)
Comment (jinalviranii) [reply jsconfig.json]: Removed

File: dockerfile (hunk 5 above)
Comment (jinalviranii) [reply dockerfile]: Done

File: src/tests/__tests__/[...params].test.js (hunk 12 above)
Comment (jinalviranii) [reply src/tests/__tests__/[...params].test.js]: Done

File: index.js (hunk 2 above)
Comment (jinalviranii) [reply index.js]: Done

File: src/app/layout.js (hunk 8 above)
Comment (jinalviranii) [reply src/app/layout.js]: Removed

File: src/app/layout.js (hunk 9 above)
Comment (jinalviranii) [reply src/app/layout.js]: Done

File: src/app/page.module.css (hunk 10 above)
Comment (jinalviranii) [reply src/app/page.module.css]: Done

File: src/pages/style/index.css (hunk 11 above)
Comment (jinalviranii) [reply src/pages/style/index.css]: Done

File: src/app/globals.css (hunk 7 above)
Comment (jinalviranii) [reply src/app/globals.css]: Done

File: src/pages/company/[...params].js [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+
+// Base URL for API requests
+const API_DOMAIN = process.env.EXTENSION_CLUSTER_URL || 'https://api.fynd.com';
+
+// HomePage component to display products
+const HomePage = ({ products = [] }) => {
+  const router = useRouter();
+  const { query } = router;
+  const applicationId = query.params[2] || null; // Extract optional application ID from query
+  const documentationUrl = 'https://api.fynd.com';
```
Comment (vivek-gofynd) [inline src/pages/company/[...params].js]: can we move const on top ? @jinalviranii

File: src/pages/company/[...params].js [hunk 14]
```diff
@@ -0,0 +1,144 @@
...
+import Image from 'next/image';
+
+// Base URL for API requests
+const API_DOMAIN = process.env.EXTENSION_CLUSTER_URL || 'https://api.fynd.com';
+
+// HomePage component to display products
+const HomePage = ({ products = [] }) => {
+  const router = useRouter();
+  const { query } = router;
+  const applicationId = query.params[2] || null; // Extract optional application ID from query
```
Comment (vivek-gofynd) [inline src/pages/company/[...params].js]: @jinalviranii 
Why are we accessing query params like this `query.params[2] ` ?
Can't we directly access like this ? const { name } = router.query; ?

File: src/pages/company/[...params].js [hunk 15]
```diff
@@ -0,0 +1,144 @@
...
+      </div>
+    </div>
+  );
+};
+
+// Fetch server-side props for the HomePage
+export async function getServerSideProps(context) {
+  try {
+    const { params } = context;
+    const companyId = params.params[0];  // First segment is always the company ID
```
Comment (vivek-gofynd) [inline src/pages/company/[...params].js]: Why we are accessing path params using array index ? @jinalviranii

File: src/pages/company/[...params].js [hunk 16]
```diff
@@ -0,0 +1,144 @@
...
+
+  // Construct the appropriate documentation link
+  const getDocumentPageLink = () => {
+    return documentationUrl
+      .replace('api', 'partners')
+      .concat(isApplicationLaunch() ? DOC_APP_URL_PATH : DOC_URL_PATH);
+  };
+
+  // Get product profile image or default image if not available
+  const getProductProfileImage = (media) => {
```
Comment (vivek-gofynd) [inline src/pages/company/[...params].js]: Are we getting any case we are not receiving image from API, if not please remove ? @jinalviranii

File: src/pages/company/[...params].js (hunk 13 above)
Comment (jinalviranii) [reply src/pages/company/[...params].js]: Moved to top

File: src/pages/company/[...params].js (hunk 14 above)
Comment (jinalviranii) [reply src/pages/company/[...params].js]: Updated as per discussion

File: src/pages/company/[...params].js (hunk 15 above)
Comment (jinalviranii) [reply src/pages/company/[...params].js]: Updated as per discussion

File: src/pages/company/[...params].js (hunk 16 above)
Comment (jinalviranii) [reply src/pages/company/[...params].js]: This is needed.

File: server.js [hunk 17]
```diff
@@ -0,0 +1,212 @@
...
+// OAuth authentication route
+app.get('/fp/install', (req, res, next) => {
+    const state = Buffer.from(JSON.stringify("abcefg")).toString('base64');
+    passport.use(getOAuthStrategy(req.query.company_id, req.query.application_id));
+    const authenticator = passport.authenticate('oauth2', { scope: [], state });
+    authenticator(req, res, next);
+});
+
+// Function to query event details
+async function queryEventDetails(accessToken) {
```
Comment (vivek-gofynd) [inline server.js]: Please change name of function @jinalviranii

File: server.js [hunk 18]
```diff
@@ -0,0 +1,212 @@
...
+    // Redirect based on company or application ID
+    const redirectUrl = req.query.application_id
+        ? `${process.env.EXTENSION_BASE_URL}/company/${req.query.company_id}/application/${req.query.application_id}`
+        : `${process.env.EXTENSION_BASE_URL}/company/${req.query.company_id}`;
+
+    return res.redirect(301, redirectUrl);
+});
+
+// API route to fetch access token
+app.get('/api/token', async(req, res) => {
```
Comment (vivek-gofynd) [inline server.js]: Make sure this routes are protected @jinalviranii

File: server.js [hunk 19]
```diff
@@ -0,0 +1,212 @@
...
+    const response = await fetch(
+        `${EXTENSION_CLUSTER_URL}/service/platform/webhook/v1.0/company/${companyId}/subscriber`,
+        requestOptions
+    );
+    return response.json(); // Return subscriber config data
+}
+
+// OAuth callback route
+app.get('/fp/auth', passport.authenticate('oauth2', { failureRedirect: '/' }), async (req, res) => {
+    const accessToken = await sQLiteStorageInstace.get(process.env.EXTENSION_API_KEY);
```
Comment (vivek-gofynd) [inline server.js]: Wrap this inside try/catch block @jinalviranii

File: server.js [hunk 20]
```diff
@@ -0,0 +1,212 @@
...
+        res.status(404).json({ message: 'No access token available. Authenticate first.' });
+    }
+});
+
+app.post('/fp/uninstall', (req, res) => {
+    // Write your cleanup logic to be executed after the extension is uninstalled.
+    res.json({ success: true });
+});
+
+app.post('/ext/webhook', (req, res) => {
```
Comment (vivek-gofynd) [inline server.js:175]: Make sure you add how to check webhook call is from legit source @jinalviranii

File: server.js [hunk 21]
```diff
@@ -0,0 +1,310 @@
...
+                    type: 'hmac',
+                    secret: process.env.EXTENSION_API_SECRET,
+                },
+                email_id: 'dev@gofynd.com',
+                event_id: eventIds,
+                status: 'active',
+            }),
+        };
+
+        const response = await fetch(
```
Comment (vivek-gofynd) [inline server.js:244]: @jinalviranii  can we use this v2 API here? https://partners.fynd.com/help/docs/sdk/latest/platform/company/webhook#registerSubscriberToEventV2

**Pull request merged**